celery/bin/base.py

```diff
@@ -1,680 +1,229 @@
+            return self.OK, self.pretty_list(n)
+        if isinstance(n, dict):
+            if 'ok' in n or 'error' in n:
+                return self.pretty_dict_ok_error(n)
+            else:
+                s = json.dumps(n, sort_keys=True, indent=4)
+                if not self.no_color:
+                    s = highlight(s, LEXER, FORMATTER)
+                return self.OK, s
+        if isinstance(n, str):
+            return self.OK, n
+        return self.OK, pformat(n)
 
-    def add(self, cls, name):
-        heappush(self.names, name)
-        self.register(cls, name=name)
+    def pretty_list(self, n):
+        if not n:
+            return '- empty -'
+        return '\n'.join(
+            f'{self.style("*", color="white")} {item}' for item in n
+        )
 
-    def load(self):
-        for name, cls in imports.load_extension_classes(self.namespace):
-            self.add(cls, name)
-        return self.names
+    def pretty_dict_ok_error(self, n):
+        try:
+            return (self.OK,
+                    text.indent(self.pretty(n['ok'])[1], 4))
+        except KeyError:
+            pass
+        return (self.ERROR,
+                text.indent(self.pretty(n['error'])[1], 4))
 
+    def say_chat(self, direction, title, body='', show_body=False):
+        if direction == '<-' and self.quiet:
+            return
+        dirstr = not self.quiet and f'{self.style(direction, fg="white", bold=True)} ' or ''
+        self.echo(f'{dirstr} {title}')
+        if body and show_body:
+            self.echo(body)
 
-class Command:
-    """Base class for command-line applications.
 
-    Arguments:
-        app (Celery): The app to use.
-        get_app (Callable): Fucntion returning the current app
-            when no app provided.
-    """
+class CeleryOption(click.Option):
+    """Customized option for Celery."""
 
-    Error = Error
-    UsageError = UsageError
-    Parser = argparse.ArgumentParser
+    def get_default(self, ctx):
+        if self.default_value_from_context:
+            self.default = ctx.obj[self.default_value_from_context]
+        return super(CeleryOption, self).get_default(ctx)
 
-    #: Arg list used in help.
-    args = ''
+    def __init__(self, *args, **kwargs):
+        """Initialize a Celery option."""
+        self.help_group = kwargs.pop('help_group', None)
+        self.default_value_from_context = kwargs.pop('default_value_from_context', None)
+        super(CeleryOption, self).__init__(*args, **kwargs)
 
-    #: Application version.
-    version = VERSION_BANNER
 
-    #: If false the parser will raise an exception if positional
-    #: args are provided.
-    supports_args = True
+class CeleryCommand(click.Command):
+    """Customized command for Celery."""
 
-    #: List of options (without preload options).
-    option_list = None
+    def format_options(self, ctx, formatter):
+        """Write all the options into the formatter if they exist."""
+        opts = OrderedDict()
```

Comment:
as far I can recall python 3.6 made Dict Orderable?

Reply:
It's an implementation detail in 3.6. Pypy happens to be that way, too. 3.7 it's official. But yes, I think the risk is low if we're 3.6+ and we don't intend to officially support any other distributions.

Reply:
I'd rather just keep it.
It may change in the future or may not. We just don't know yet.